### PR TITLE
CAMEL-12183: add support for Wordpress REST API

### PR DESCRIPTION
**Pull request by ricardozanini**
PR for Wordpress component. I've tried to stick to the project standards, but I may missed something. Also, the parent project `component` may need to add this project as a module. :)

### CODE REVIEW

File: components/camel-wordpress/src/main/docs/wordpress-component.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    configuration.setApiVersion("2");
+    configuration.setUrl("http://yoursite.com/wp-json/");
+    component.setConfiguration(configuration);
+    getContext().addComponent("wordpress", component);
+    
+    from("wordpress:post?id=1")
+      .to("mock:result");
+}
+--------------------------------------------------
+==== Path Parameters
```
Comment (oscerd) [inline components/camel-wordpress/src/main/docs/wordpress-component.adoc]: Like the other components you need to add the placeholders for component options and endpoint options

// component options: START
// component options: END

// endpoint options: START
//endpoint options: END

The docs will be generated automatically

File: components/camel-wordpress/src/main/java/org/apache/camel/component/wordpress/config/WordpressEndpointConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+import org.apache.camel.util.IntrospectionSupport;
+import org.wordpress4j.model.SearchCriteria;
+
+@UriParams
+public class WordpressEndpointConfiguration extends WordpressComponentConfiguration {
```
Comment (oscerd) [inline components/camel-wordpress/src/main/java/org/apache/camel/component/wordpress/config/WordpressEndpointConfiguration.java]: Is there a particular reason to have two different configuration?

**Comment by oscerd** [comment]
After that we will need a Karaf feature for the component, but we may take care of that after.

@davsclaus can you please add your feedback too? Thanks.

**Comment by oscerd** [comment]
Also, maybe it would be better to squash all the commit in one and add a reference in the commit message to the JIRA number CAMEL-12183

Lastly, take a look at the codestyle by running the -Psourcecheck and fixing possible errors

**Comment by onders86** [comment]
for org.wordpress4j.* package, is it a part of component or an external API? I am not sure if it is ok to have it placed under camel repository if it is an external one?

**Comment by oscerd** [comment]
I didn't find references of wordpress4j in central

**Comment by oscerd** [comment]
By the way the package should be renamed to something under org.apache.camel. It doesn't seem to be an external one, at least I can't find source.

**Comment by ricardozanini** [comment]
Hi guys!

Many thanks for your help!

Just clean up with all your suggestions, but let me clarify my choices:

1) Two configuration classes were just a design choice, but thinking again about it, keep only one class makes a lot of sense.
2) I had an intention to maintain the org.wordpress4j from my own, but would be too much work (and error prune) to do that, so I've decided to keep it together, but with limit boundaries between then. If this component grows, I may suggest in the future for us to do so (many components relies on this like Facebook and Twitter).
3) How do I squash all the commit and link it again to the JIRA? Could you help me accomplish that?

Thanks again!

**Comment by oscerd** [comment]
Is the code of org.wordpress4j maintained somewhere? What license to they have? Is this stuff released on some repository? I'm guess you can't copy code from other libraries in an ASF project.

**Comment by ricardozanini** [comment]
Oh god! No it’s mine! My intention was to create a distinct project to handle the Wordpress integration just like Facebook and Twitter components do. The domain is registered in my name and the Girhub project (its empty) too.

**Comment by oscerd** [comment]
Ok, great :-) 

To squash the commit, first:

git pull --rebase origin master

if origin is your apache remote repo's name

then

git rebase -i HEAD~9

you'll get a list of commits

leave pick <commit_hash> on the first one

for the others change "pick" to "squash"

save and exit.

At the end you'll get a long commit message, before saving, remove the commit message and just add "CAMEL-12183: add support for Wordpress REST API"

Maybe do this on a new branch, so you can always take the original branch back and start again.

**Comment by oscerd** [comment]
Maybe it's better to open another PR.

**Pull request closed without merging**